### DISCUSSION TITLE
Is there a way to cancel a repeatable job from within the worker

### DISCUSSION THREAD

**Discussion by alanwguo** [taskforcesh/bullmq#2072] [Q&A]
Is there an easy way to cancel a repeating job from within a worker that is processing a job.

I see `job.disard()` is available but the docs only mention it prevents retries from failures.

What I would like is something like `job.discard()` that tells the itself to stop repeating.

**Comment by manast** [comment]
I think you would need something like this: https://api.docs.bullmq.io/classes/v4.Queue.html#removeRepeatableByKey
